src/VerifiableFactory.sol

```diff
@@ -74,26 +72,32 @@ contract VerifiableFactory {
             return false;
         }
         try IProxy(proxy).salt() returns (uint256 salt) {
-            try IProxy(proxy).creator() returns (address creator) {
-                // verify the creator matches this factory
-                if (address(this) != creator) {
-                    return false;
-                }
+            try IProxy(proxy).owner() returns (address owner) {
+                try IProxy(proxy).creator() returns (address creator) {
+                    return _verifyContract(proxy, creator, owner, salt);
+                } catch {}
+            } catch {}
+        } catch {}
+
+        return false;
+    }
 
-                // reconstruct the address using CREATE2 and verify it matches
-                bytes32 outerSalt = keccak256(abi.encode(msg.sender, salt));
+    function _verifyContract(address proxy, address creator, address owner, uint256 salt) private view returns (bool) {
+        // verify the creator matches this factory
```

Comment:
Is there any need for this check, or for the proxy to be able to return the creator? If it doesn't match, the address will not validate.

Reply:
oh right, since creator is the part of the address creation, this is unnecessary. gonna remove it.